.github/workflows/ci.yml

```diff
@@ -31,6 +31,22 @@ jobs:
       SCARB_VERSION: 0.6.0
     runs-on: ubuntu-latest
     steps:
+      # extract branch name
+      - name: Extract branch name
+        if: github.event_name != 'pull_request'
+        shell: bash
+        run: echo "BRANCH_NAME=$(echo ${GITHUB_REF#refs/heads/})" >> $GITHUB_ENV
+        id: extract_branch
+
+      # extract branch name on pull request
+      - name: Print branch name
+        if: github.event_name == 'pull_request'
+        run: echo "BRANCH_NAME=$(echo $GITHUB_HEAD_REF)" >> $GITHUB_ENV
```

Comment:
You can verify it, but `github.ref_name` https://docs.github.com/en/actions/learn-github-actions/contexts should work without writing specific code for PRs and other events.

Reply:
https://github.com/foundry-rs/starknet-foundry/actions/runs/5903277226/job/16012867783 I tested `GITHUB_REF_NAME ` has the same definition and it doesn't get the name of the branch with processing code
![Zrzut ekranu 2023-08-25 o 12 15 07](https://github.com/foundry-rs/starknet-foundry/assets/15279784/f517f883-fb49-4fc8-ae03-f1f257be5899)
under GITHUB_SHA if value of `GITHUB_REF_NAME`